crates/transaction-pool/src/traits.rs

```diff
@@ -769,6 +768,9 @@ pub trait PoolTransaction:
     /// Associated type representing the recovered pooled variant of the transaction.
     type Pooled: Into<Self>;
 
+    /// Define a method to convert from the `Pooled` type to `Self`
+    fn from_pooled(pooled: Self::Pooled) -> Self;
```

Comment:
I think this is also a temporary, thing, because there can be a usecase where this conversion requires additional context, but this is a good solution for now